src/Native/Runtime/thread.cpp

```diff
@@ -1295,9 +1295,15 @@ EXTERN_C NOINLINE void FASTCALL RhpReversePInvokeAttachOrTrapThread2(ReversePInv
 // PInvoke
 //
 
+extern "C" int InitializeRuntime();
+
 // Standard calling convention variant of RhpReversePInvoke
 COOP_PINVOKE_HELPER(void, RhpReversePInvoke2, (ReversePInvokeFrame * pFrame))
 {
+    // Entrypoint for reverse PInvoke
+    // ensure runtime is initialized
+    InitializeRuntime();
```

Comment:
This is a hot fast path. The runtime initialization should to be done in `RhpReversePInvokeAttachOrTrapThread2` - similar to how the thread initialization is done.